src/main/java/com/ft/methodeimagemodelmapper/service/MethodeImageModelMapper.java

```diff
@@ -164,7 +167,24 @@ private boolean isEomTypeSupported(final EomFile eomFile) {
                 .withCanBeDistributed(canBeDistributed)
                 .withCanBeSyndicated(canBeSyndicated)
                 .withRightsGroup(rightsGroup)
-                .withMasterSource(fotowareID);
+                .withMasterSource(fotowareID)
+                .withExternalBinaryUrl(externalBinaryUrl);
+    }
+
+    private String resolveExternalBinaryUrl(EomFile eomFile, String transactionId, XPath xpath, Document attributesDocument) throws XPathExpressionException {
+        String externalBinaryUrl = xpath.evaluate("/meta/picture/ExternalUrl", attributesDocument);
+        boolean matched = false;
+        for (final String sample : externalBinaryUrlWhitelist) {
+            if (externalBinaryUrl.matches(sample)) {
+                matched = true;
+                LOGGER.info("This image will be assigned an externalBinaryUrl from a custom set location. externalBinaryUrl={} transaction_id={}", externalBinaryUrl, transactionId);
+                break;
```

Comment:
You could just return the `externalBinaryUrl` here and remove the `matched` boolean.

Reply:
simplified logic